### PR TITLE
Add metadata to ProductMedia.

### PR DESCRIPTION
**Pull request by zedzior**
https://github.com/saleor/saleor/issues/11834
I want to merge this change because I want to have metadata stored for images.

Example of use case:
- Provide image labels
- Set display preference

# Impact

* [x] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/product/migrations/0180_productmedia_metadata.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            model_name="productmedia",
+            name="private_metadata",
+            field=models.JSONField(
+                blank=True,
+                default=dict,
+                encoder=saleor.core.utils.json_serializer.CustomJsonEncoder,
+                null=True,
+            ),
+        ),
+    ]
```
Comment (zedzior) [inline saleor/product/migrations/0180_productmedia_metadata.py:34]: Pre-commit's `migration-check` doesn't allow me to place GIN index here

Comment (zedzior) [reply saleor/product/migrations/0180_productmedia_metadata.py:34]: Thanks!

File: saleor/graphql/schema.graphql [hunk 2]
```diff
@@ -6518,8 +6518,54 @@ type ProductImage {
...
+  Added in Saleor 3.3.
+  
+  Note: this API is currently in Feature Preview and can be subject to changes at later point.
+  """
+  privateMetafield(key: String!): String
+
+  """
+  Private metadata. Requires staff permissions to access. Use `keys` to control which fields you want to include. The default is to include everything.
+  
+  Added in Saleor 3.3.
```
Comment (korycins) [inline saleor/graphql/schema.graphql:6541]: For some reason you have incorrect label here. We need to add here a proper labels, also for some fields `added_in_3x` is missing (like `privateMetadata: [MetadataItem!]!`)

Comment (zedzior) [reply saleor/graphql/schema.graphql:6541]: There is an issue for that: https://github.com/saleor/saleor/issues/11702. WIll add something temporary to fix it. Thanks!

**Review by fowczarek — COMMENTED** [review]
Can you add tests for mutations?

**Review by maarcingebala — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**